### PR TITLE
bug fix while removing object from queue

### PR DESCRIPTION
**Pull request by anirudhramanan**

### CODE REVIEW

File: batching/src/main/java/com/flipkart/batching/tape/InMemoryObjectQueue.java [hunk 1]
```diff
@@ -64,6 +69,15 @@ public void remove(int n) {
         }
     }
 
+    private void removeFromQueue() {
+        if (tasks.isEmpty()) {
+            LogUtil.log(TAG, "The queue is empty");
+            return;
+        }
+
+        tasks.remove();
```
Comment (anirudhramanan) [inline batching/src/main/java/com/flipkart/batching/tape/InMemoryObjectQueue.java]: Else we could catch the NoSuchElementException exception here.

Comment (anirudhramanan) [inline batching/src/main/java/com/flipkart/batching/tape/InMemoryObjectQueue.java:78]: Else we could catch the NoSuchElementException exception here.

File: batching/src/test/java/com/flipkart/batching/tape/InMemoryObjectQueueTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        arrayList.add("test4");
+
+        for (String sample : arrayList) {
+            inMemoryObjectQueue.add(sample);
+        }
+
+        //remove all the 4 elements added to the queue
+        inMemoryObjectQueue.remove(arrayList.size());
+
+        //try removing another element from the queue.
```
Comment (thekirankumar) [inline batching/src/test/java/com/flipkart/batching/tape/InMemoryObjectQueueTest.java:40]: Write a comment saying this should no throw nosuchelement exception

File: batching/src/test/java/com/flipkart/batching/tape/InMemoryObjectQueueTest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+@RunWith(RobolectricGradleTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21)
+public class InMemoryObjectQueueTest {
+
+    /**
+     * Test for {@link InMemoryObjectQueue#remove()} )
+     */
+    @Test
+    public void testRemove() {
```
Comment (thekirankumar) [inline batching/src/test/java/com/flipkart/batching/tape/InMemoryObjectQueueTest.java:24]: The test name should convey what you expect out of this. Eg testRemoveLastDoesntThrow

**Pull request merged**